catboost/private/libs/hyperparameter_tuning/hyperparameter_tuning.cpp

```diff
@@ -687,6 +687,23 @@ namespace {
         oneIterLogger.OutputParameters(parametersToken, jsonParams);
     }
 
+    bool ParseJsonParams(
+        const NJson::TJsonValue& modelParamsToBeTried,
+        NCatboostOptions::TCatBoostOptions *catBoostOptions,
+        NCatboostOptions::TOutputFilesOptions *outputFileOptions) {
+        try {
+            NJson::TJsonValue jsonParams;
+            NJson::TJsonValue outputJsonParams;
+            NCatboostOptions::PlainJsonToOptions(modelParamsToBeTried, &jsonParams, &outputJsonParams);
+            *catBoostOptions = NCatboostOptions::LoadOptions(jsonParams);
+            outputFileOptions->Load(outputJsonParams);
```

Comment:
Validation is already done inside `LoadOptions()` and `Load()` methods, so I've decided to avoid parsing and validating parameters twice (the case of `IsValidParams(modelParamsToBeTried)` method) and just catch an exception where appropriate.